File: mobile/lib/features/feasibility/domain/entities/approval_step.dart
class ApprovalStep {
  ApprovalStep({
    required this.id,
    required this.approverName,
    required this.order,
    required this.decision,
    required this.comment,
    required this.decidedAtUtc,
  });

  final String id;
  final String approverName;
  final int order;
  final String decision; // Pending, Approved, Rejected
  final String? comment;
  final DateTime? decidedAtUtc;

  factory ApprovalStep.fromJson(Map<String, dynamic> json) {
    return ApprovalStep(
      id: json['id'].toString(),
      approverName: json['approverName'] as String? ?? '',
      order: (json['order'] as num?)?.toInt() ?? 0,
      decision: json['decision'] as String? ?? 'Pending',
      comment: json['comment'] as String?,
      decidedAtUtc: json['decidedAtUtc'] == null ? null : DateTime.tryParse(json['decidedAtUtc'] as String),
    );
  }
}

File: backend/src/Services/FeasibilityService/Ozdilek.PM.FeasibilityService.Domain/ApprovalStep.cs
using Ozdilek.PM.SharedKernel.Exceptions;
using Ozdilek.PM.SharedKernel.Persistence;

namespace Ozdilek.PM.FeasibilityService.Domain;

public class ApprovalStep : BaseEntity
{
    public Guid FeasibilityItemId { get; private set; }
    public string ApproverName { get; private set; } = string.Empty;
    public int Order { get; private set; }
    public ApprovalDecision Decision { get; private set; } = ApprovalDecision.Pending;
    public string? Comment { get; private set; }
    public DateTimeOffset? DecidedAtUtc { get; private set; }

    private ApprovalStep() { }

    public ApprovalStep(Guid feasibilityItemId, string approverName, int order)
    {
        FeasibilityItemId = feasibilityItemId;
        ApproverName = approverName;
        Order = order;
    }

    public void Decide(bool approve, string? comment)
    {
        if (Decision != ApprovalDecision.Pending)
        {
            throw new DomainException("Bu onay adımı zaten karara bağlanmış.");
        }

        Decision = approve ? ApprovalDecision.Approved : ApprovalDecision.Rejected;
        Comment = comment;
        DecidedAtUtc = DateTimeOffset.UtcNow;
    }
}
